New "Learning Environment" Page (#49299)
kubernetes/website: content/en/docs/setup/learning-environment/_index.md

---
title: Learning environment
weight: 20
description: Tools to try out Kubernetes locally
no_list: true
---
<!-- overview -->

This page outlines some tools that you can use to set up a local Kubernetes cluster to try out Kubernetes concepts and features.
Before diving into setup, it's highly recommended to familiarize yourself with core Kubernetes concepts. You can find these in the [Concepts](/docs/concepts) section.

<!-- body -->

## How to communicate with a Kubernetes cluster?
When working with Kubernetes, you need a way to interact with and manage your cluster. This involves tasks such as deploying applications, inspecting resources, and debugging issues. The `kubectl` command-line tool serves as the primary interface for communicating with a configured Kubernetes cluster, allowing you to send commands and receive responses to and from the cluster's API server.

> 

Reviewer: ```suggestion

You'll need to [install kubectl](/docs/tasks/tools/#kubectl) unless you choose a learning environment
that provides it for you.

```

For more information including a complete list of kubectl operations, see the [`kubectl` reference documentation](/docs/reference/kubectl).

kubectl is installable on a variety of Linux platforms, macOS and Windows. Find your preferred operating system [here](/docs/tasks/tools/#kubectl).

## Options for trying out Kubernetes locally
To experiment with Kubernetes locally, you need tools that allow you to simulate a Kubernetes cluster on your machine. These tools create lightweight clusters for testing, development, or learning purposes without requiring extensive infrastructure.

> ### 1. kind

Reviewer: As a project, we recommend minikube over kind. kind was originally a tool for development rather than learning (but is still a viable option).

You could also make a heading that combines them, listing them in alphabetical order; that is:
```suggestion
### 1. kind and minikube {#kind-and-minikube}

```

[`kind`](https://kind.sigs.k8s.io/) lets you run Kubernetes on your local computer. This tool requires that you have either [Docker](https://www.docker.com/) or [Podman](https://podman.io/) installed.

The kind [Quick Start](https://kind.sigs.k8s.io/docs/user/quick-start/) page shows you what you need to do to get up and running with kind.

### 2. minikube
Like `kind`, [`minikube`](https://minikube.sigs.k8s.io/) is a tool that lets you run Kubernetes locally. `minikube` runs an all-in-one or a multi-node local Kubernetes cluster on your personal computer (including Windows, macOS and Linux PCs) so that you can try out Kubernetes, or for daily development work.

You can follow the official [Get Started](https://minikube.sigs.k8s.io/docs/start/) guide to set it up.

> ### 3. Other 3rd party tools

Reviewer: ```suggestion
### 3. Other 3rd party tools {#third-party-tools}

```
Avoid putting the number into the fragment identifier.

> There are additional tools like [MicroK8s](https://microk8s.io/docs) and [k3d](https://k3d.io/stable/) for running Kubernetes clusters locally.

Reviewer: This needs a _third party content_ marker. How about:
```suggestion

{{% thirdparty-content %}}

There are additional tools such as [MicroK8s](https://microk8s.io/docs) and [k3d](https://k3d.io/stable/) for running Kubernetes clusters locally. You can use the [CNCF Landscape(https://landscape.cncf.io/)
to find a tool that for deploying certified Kubernetes locally.
```

> ### 4. Online Playground Environment

Reviewer: (nit)
```suggestion
### 4. Online playground environments {#online-playgrounds}

```

If you don’t want to install Kubernetes locally, you can use online environments to practice.

> - [Killer Coda](https://killercoda.com/): Interactive Kubernetes labs for various scenarios.

Reviewer: ```suggestion
- [Killercoda](https://killercoda.com/): Interactive Kubernetes labs for various scenarios.
```

- [Play With Kubernetes](https://labs.play-with-k8s.com/): Browser-based Kubernetes playground to spin up clusters on demand.

> ## What's next: Practice setting up a production-like cluster

Reviewer: Try this:
```suggestion
## {{% heading "whatsnext" %}}

### Install local tools

- [Install kubectl](/docs/tasks/tools/#kubectl) if you didn't already do that
  -  and you can [add plugins to kubectl](/docs/tasks/extend-kubectl/kubectl-plugins/#installing-kubectl-plugins) if you'd like to
- You will also need a `.kubeconfig` configuration file

### Tutorials

Try a [tutorial](/docs/tutorials/)  to learn more about Kubernetes.

### Training

Look into the [training](/training/) that's available to help you learn Kubernetes.

### Practice setting up a production-like cluster

```

Also see https://github.com/kubernetes/website/issues/38681 - we don't yet have a task page about setting up kubeconfig. Once we do, other pages can link to it.

While the tools above are great for learning, setting up a production-like cluster provides deeper insights into how Kubernetes operates in real-world scenarios. 

> [Kubeadm]() is a tool designed to help you set up a secure and functional cluster with minimal complexity.

Reviewer: ```suggestion
The `kubeadm` tool is designed to help you [set up a production-grade cluster](/docs/setup/production-environment/tools/kubeadm/) with minimal complexity.

Once you've deployed a baseline cluster with kubeadm, typically you'll want to customize it or extend it
to deliver a cloud-native platform.
Read [production considerations](/docs/setup/production-environment/#production-considerations)
when you're ready to learn more.
```
